File: src/main/java/architecture/goldenboughs_lib/api/world/entity/IBehaviorTreeMob.java
package architecture.goldenboughs_lib.api.world.entity;

import architecture.goldenboughs_lib.api.world.entity.ai.behavior.BTRoot;
import net.minecraft.world.entity.Mob;

public interface IBehaviorTreeMob<T extends Mob> {
	BTRoot<T> createBehaviorTree();
}

File: src/main/kotlin/architecture/goldenboughs_lib/api/world/entity/IBehaviorTreeMob.kt
package architecture.goldenboughs_lib.api.world.entity

import architecture.goldenboughs_lib.api.world.entity.ai.behavior.BTRoot
import net.minecraft.world.entity.Mob

interface IBehaviorTreeMob<T : Mob> {
	fun createBehaviorTree(): BTRoot<T>
}
